metal/label_model/class_balance.py

```diff
@@ -0,0 +1,136 @@
+from itertools import product
+
+import numpy as np
+import torch
+from torch import nn, optim
+
+
+class ClassBalanceModel(nn.Module):
+    """A model for learning the class balance, P(Y=y), given a subset of LFs
+    which are *conditionally independent*, i.e. \lambda_i \perp \lambda_j | Y,
+    for  i != j.
+
+    Learns the model using a tensor factorization approach.
+
+    Note: This approach can also be used for estimation of the LabelModel, may
+    want to later refactor and expand this class.
+    """
+
+    def __init__(self, k, abstains=True, config=None):
+        super().__init__()
+        self.config = config
+        self.k = k
+        self.abstains = abstains
+        self.k_0 = 0 if self.abstains else 1
+        self.k_lf = k + 1 if self.abstains else k
+
+        # Estimated quantities (np.array)
+        self.cond_probs = None
+        self.class_balance = None
+
+    def _get_overlaps_tensor(self, L):
+        """Transforms the input label matrix to a three-way overlaps tensor.
+
+        Args:
+            L: (np.array) An n x m array of LF output labels, in {0,...,k} if
+                self.abstains, else in {1,...,k}, generated by m conditionally
+                independent LFs on n data points
+
+        Outputs:
+            O: (torch.Tensor) A (m, m, m, k, k, k) tensor of the label-specific
```

Comment:
Wow--a 6D tensor. May be worth saying something like, "for example, O[a,b,c,d,e,f] corresponds to...[explanation that uses a-f]"

Reply:
Done!